.appveyor/appveyor_helper.bat

```diff
@@ -0,0 +1,40 @@
+@echo off
+setlocal enableDelayedExpansion
+
+rem for /f %%f in ('wmic cpu get NumberOfLogicalProcessors /value ^| find "="') do set %%f
+rem set /a "ncpus=%NumberOfLogicalProcessors%*%NUMBER_OF_PROCESSORS%"
+set /a "started=0, ended=0"
+set hasmore=1
+1>nul 2>nul del *.out *.passed
+for /f "usebackq" %%A in (`dir /on /b .appveyor ^| findstr /B /R "^build_.*\.bat$"`) do (
+    if !started! lss %NUMBER_OF_PROCESSORS% (
```

Comment:
(If you want I can put comments in code also)
Actually this script is just helper script to build PBS deps in parallel to reduce time of building all PBS deps.
Now on to your question regarding how this script works:
1. it will list all build_*.bat file in .appveyor directory
2. one by one it will execute each batch file in background while capturing output and error or batchfile in \<batch file name>.bat.out file in current directory
3. while executing batch file it will limit number of started background process to number of processors available in system.
4. once it started background process reached limit it goes to wait loop where it will wait for atleast one background process to complete by looking at exclusive lock on \<batch file name>.bat.out created by start of process and that lock will be released as soon as process ends
5. as soon as one processor ends it checks for \<batch file name>.bat.passed file if it exists that means background process completed successfully else some error occurred. Now in case of error this script will show contents of \<batch file name>.bat.out to console else it will just put one line saying \<batch file name>.bat finished.
6. Now since one process if finished we have one room to start new process so it will go back to main loop to start new background process for building next PBS deps and main loop again goes to wait loop and this goes on till end of all batch file in .appveyor directory

Hope you understand how this script works! Please do not hesitate to ask question if you don't 😄